drivers/i2s/i2s_mcux_sai.c

```diff
@@ -1256,30 +1312,30 @@ static const struct i2s_driver_api i2s_mcux_driver_api = {
 									\
 	static const struct i2s_mcux_config i2s_##i2s_id##_config = {	\
 		.base = (I2S_Type *)DT_INST_REG_ADDR(i2s_id),		\
-		.clk_src = DT_INST_PROP(i2s_id, clock_mux),		\
-		.clk_pre_div = DT_INST_PROP(i2s_id, pre_div),		\
-		.clk_src_div = DT_INST_PROP(i2s_id, podf),		\
+		.clk_src = DT_INST_PROP_OR(i2s_id, clock_mux, 0),		\
+		.clk_pre_div = DT_INST_PROP_OR(i2s_id, pre_div, 0),		\
+		.clk_src_div = DT_INST_PROP_OR(i2s_id, podf, 0),		\
 		.pll_src =						\
-			DT_PHA_BY_NAME(DT_DRV_INST(i2s_id),		\
-				pll_clocks, src, value),		\
+			DT_PHA_BY_NAME_OR(DT_DRV_INST(i2s_id),		\
+				pll_clocks, src, value, 0),		\
 		.pll_lp =						\
-			DT_PHA_BY_NAME(DT_DRV_INST(i2s_id),		\
-				pll_clocks, lp, value),			\
+			DT_PHA_BY_NAME_OR(DT_DRV_INST(i2s_id),		\
+				pll_clocks, lp, value, 0),			\
 		.pll_pd =						\
-			DT_PHA_BY_NAME(DT_DRV_INST(i2s_id),		\
-				pll_clocks, pd, value),			\
+			DT_PHA_BY_NAME_OR(DT_DRV_INST(i2s_id),		\
+				pll_clocks, pd, value, 0),			\
 		.pll_num =						\
-			DT_PHA_BY_NAME(DT_DRV_INST(i2s_id),		\
-				pll_clocks, num, value),		\
+			DT_PHA_BY_NAME_OR(DT_DRV_INST(i2s_id),		\
+				pll_clocks, num, value, 0),		\
 		.pll_den =						\
```

Comment:
Looks like a default has been added to all DT properties. I would leave the one's not needed for this support unchanged.